packages/contracts/tests/python/test.py

```diff
@@ -2624,13 +2626,60 @@ def test_too_soon_bonds_under_unrevealed_commit(self):
         
         self.assertEqual(self.rc0.functions.balanceOf(k3).call(), claimable)
 
+    #@unittest.skipIf(WORKING_ONLY, "Not under construction")
+    def test_is_history_of_unfinalized_question_valid(self):
 
- 
+        if not IS_PROVE_HISTORY_SUPPORTED:
+            print("Skipping test_is_history_of_unfinalized_question_valid, not supported by this version")
+            return
 
+        k3 = self.web3.eth.accounts[3]
 
+        if ERC20:
+            self._issueTokens(k3, 100000, 50000)
 
+        st = None
+        st = self.submitAnswerReturnUpdatedState( st, self.question_id, 1001, 0, 20, k3)
+        st = self.submitAnswerReturnUpdatedState( st, self.question_id, 1002, 20, 40, k3)
+        st = self.submitAnswerReturnUpdatedState( st, self.question_id, 1001, 40, 80, k3)
+        st = self.submitAnswerReturnUpdatedState( st, self.question_id, 1004, 80, 160, k3)
+        st = self.submitAnswerReturnUpdatedState( st, self.question_id, 1003, 160, 320, k3)
+        st = self.submitAnswerReturnUpdatedState( st, self.question_id, 1001, 320, 640, k3)
 
+        hashes = st['hash']
+        addrs = st['addr']
+        bonds = st['bond']
+        answers = st['answer']
 
+        self.assertTrue(self.rc0.functions.isHistoryOfUnfinalizedQuestionValid(self.question_id, hashes, addrs, bonds, answers).call())
+
+        bad_bonds = bonds.copy()
+        bad_bonds[1] = 987
+        self.assertFalse(self.rc0.functions.isHistoryOfUnfinalizedQuestionValid(self.question_id, hashes, addrs, bad_bonds, answers).call())
```

Comment:
you could also add test cases for bad_addr or bad_answers

Reply:
Done, also added some variation to the question creator addresses